addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2732,15 +2732,15 @@ var SnippetsMenu = Widget.extend({
     _computeSelectorFunctions: function (selector, exclude, target, noCheck, isChildren, excludeParent) {
         var self = this;
 
-        // TODO the `:not([contenteditable="true"])` part is designed to make
-        // images with such attribute editable even when they are in an
-        // environment where editing is not normally possible. This should be
-        // reviewed if we are to handle more hierarchy of editable nodes being
-        // editable despite their non editable environment.
+        // TODO the `:not(.o_editable_media)` part is designed to make images
+        // with that class editable even when they are in an environment where
+        // editing is not normally possible. This should be reviewed if we are
+        // to handle more hierarchy of editable nodes being editable despite
+        // their non editable environment. 
         // Without the `:not(.s_social_media)`, it is no longer possible to edit
-        // icons in the social media snippet. This should be fixed in a more
-        // proper way to get rid of this hack.
-        exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable:not(.s_social_media) :not([contenteditable="true"])`;
+        // the social media snippet. This should be fixed in a more proper way
+        // to get rid of this hack.
+        exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable:not(.s_social_media) :not(.o_editable_media)`;
```

Comment:
@dmo-odoo the social media snippet div itself is not editable so I have to leave an exception for it, unless it should also receive class `o_editable_media` (like its icons/images) ?

Reply:
I don't think the `o_editable_media` class should go on anything else than the editable media itself. Though a case could be made for it to signify that every media inside of the node is editable, I feel like this was not the original intent for that class.

I would remove the first part of the TODO because I feel like the `o_editable_media` class _is_ the "reviewed" version of the fix to the initial problem.
```js
        // The `:not(.o_editable_media)` part is designed to make images with
        // that class editable even when they are in an environment where
        // editing is not normally possible.
        // TODO: Without the `:not(.s_social_media)`, it is no longer possible
        // to edit the social media snippet. This should be fixed in a more
        // proper way to get rid of this hack.
```

However, it seems like there is still an issue for which the `.s_social_media` hack is still necessary and therefore still has its associated TODO. If the standard mechanism of `snippets.editor.js` to detect whether something should have an editor block in the sidebar or not is to check contentEditable then I guess the hack should be kept. However, what would you think @qsm-odoo if the standard mechanism would instead check contentEditable OR that you have any `.o_editable_media` as direct children (hoping those are direct children of `.s_social_media` from the top of my head) ? Because the social media snippet is basically only those inside a non-editable div.

Reply:
They're not a direct children, it goes
`<div class="s_social_media"><a class="there_is_no_class_such_as s_social_media_custom"><i class="o_editable_media>`

Reply:
Thanks for the clarification. Then my suggestion won't work, unfortunately. The problem is that we can't really make it about "if the node contains any .o_editable_media" because then all nodes from the direct parent to the root will satisfy this condition... let's wait for @qsm-odoo's input as he might have a better idea since the snippets editor are more his realm than ours.

Reply:
Not sure to understand the discussion here sorry :flushed:

`.o_not_editable:not(.s_social_media) :not([contenteditable="true"])`
- I think that once we have this new `o_editable_media` class, then the `:not(.s_social_media)` part can be removed, it was there just waiting for the `o_editable_media` class I think ? -> the `:not([contenteditable="true"])` not being enough because icon elements were void elements and were thus not able to keep `[contenteditable="true"]`.
- I think that `:not([contenteditable="true"])` should be kept, alongside the new `:not(.o_editable_media)`: an editable non-media element inside a non-editable one should have its right-panel options available since it is editable. Arguably, only its content is editable so only its children should have their options... but I think it's ok to also enable the ones of `[contenteditable="true"]` elements too (let's not create a "o_editable_element" class at least for now ^^).

Reply:
So, where should the `o_editable_media` class be applied ?
- Icons for custom social media links
- Links for custom social media (containing the icons)
- Social media snippets (divs with the `s_social_media` class), thus affecting custom and not custom children ?
- All/some of the above ?

For now, they are applied on custom social media icons. Therefore the `:not(.s_social_media)` is still necessary. The `:not([contenteditable="true"])` however is not.

Reply:
> If your task is about doing what we talked about lastly, then it should be on all media that we want to be editable even though they are in non editable environment. So here, on all social media icons (not only "custom" ones as you mentioned, we still need to let options appear for "non custom" ones?).

I'm just chiming in to say that only allowing to edit custom ones was the original spec when we discussed it in your office. It was because the other icons have their url configured in a setting. It's ok if you changed your mind but I just wanted to be sure we were not missing something here.

Reply:
> If your task is about doing what we talked about lastly, then it should be on all media that we want to be editable even though they are in non editable environment. So here, on all social media icons (not only "custom" ones as you mentioned, we still need to let options appear for "non custom" ones?).

So, I'll await a decision about the spec. My question was of a technical nature though: is the class supposed to be on the `<i>`, `<a>` or `<div>` tag ? All cases could work whatever the final spec is, but if we go with "custom icons editable but not standard icons" it would make more sense to keep the class on the `<i>` tag IMO.

> So I think you can put it on non-custom ones too and thus remove :not(.s_social_media)?

Here specifically the issue is not about custom vs non-custom icons. It's about the `o_editable_media` class being on the `<i>` nodes and thus the edition options disappearing for the social media snippet itself (not its individual links or icons) if I remove `:not(.s_social_media)`. The decision for the spec and implementation will affect whether or not I can remove some parts of this selector.

> I suggested keeping it as it makes sense generically. Not because it is needed for the s_social_media part.

Gotcha, putting it back.

Reply:
> [...] Can you test with adding o_editable_media on all icons and tell me if something is broken because of it? +1

Indeed, adding that class to all icons (not just custom ones) presents no apparent issues and actually fixes an issue I was unaware of (it re-enables the edition of _other_ aspects of standard icons, such as background color etc...). Thank you for pointing this out !

Reply:
> Why would the options disappear if you remove the :not(.s_social_media) part? The selector targets the descendants of the element holding the o_not_editable class, not the element itself... so the options on the snippet itself should stay whatever you do?

I was unaware of the space in the selector indicating parent-children relationship (unless preceded by a comma). I think the behavior is now correct, whether templates are updated or not we can select the snippet itself (not always the media). The new class is only added on the media itself.

Many, many thanks !